File: Data Structures & Algorithms/find-minimum-in-rotated-sorted-array/submission-2.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        ans=nums[0]
        def bst(low,high,ans):
            if low>high:
                return ans
            mid=int((low+high)/2)
            if ans<=nums[mid]:
                return bst(mid+1,high,ans)
            else:
                ans=nums[mid]
                return bst(low,mid-1,ans)
        return bst(0,len(nums)-1,ans)

File: Data Structures & Algorithms/jump-game/submission-2.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int last=nums.size()-1;
        for (int i=nums.size()-2; i>=0; --i) {
            if (nums[i]>=last-i) {
                last=i;
            }
        }
        if (last==0) return true;
        return false;
    }
};
